src/components/user/user.service.ts

```diff
@@ -366,6 +373,49 @@ export class UserService {
       nodevar: 'user',
     });
 
+    // Update email
+    if (input.email) {
+      // Remove old emails and relations
+      await this.db
+        .query()
+        .match([
+          node('user', ['User', 'BaseNode'], { id: user.id }),
+          relation('out', 'oldRel', 'email', { active: true }),
+          node('email', 'EmailAddress:Property'),
+        ])
+        .delete('oldRel')
+        .delete('email')
+        .run();
+
+      try {
+        const createdAt = DateTime.local();
+        await this.db
+          .query()
+          .match([node('user', ['User', 'BaseNode'], { id: user.id })])
+          .create([
+            node('user'),
+            relation('out', '', 'email', {
+              active: true,
+              createdAt,
+            }),
+            node('email', 'EmailAddress:Property', {
+              value: input.email,
+              createdAt,
+            }),
+          ])
+          .run();
```

Comment:
Actually I think this whole thing can just be merged into `sgUpdateProperties` call. You could still try/catch that to have the uniqueness check there.

Reply:
Hmm... if we merge into `sgUpdateProperties`, the `email` property will not have `EmailAddress` label. @michaelmarshall Can you confirm this part?

Reply:
Any property that has a uniqueness constraint must have a unique label. In order for email address to be unique, they must have the `EmailAddress` label.